### PR TITLE
LXC-4.x: do not drop sys_admin capability

### PR DESCRIPTION
**Pull request by stefangweichinger**
This is a draft only, to document things.
I still have to do more tests with fresh VMs, both buster and bullseye.

### CODE REVIEW

File: ansible/roles/lxc/tasks/main.yml [hunk 1]
```diff
@@ -340,7 +340,7 @@
   lineinfile:
     path: '/var/lib/lxc/{{ item.name | d(item) }}/config'
     regexp: '^lxc\.mount\s+=\s+'
-    line: 'lxc.mount = /var/lib/lxc/{{ item.name | d(item) }}/fstab'
+    line: 'lxc.mount.fstab = /var/lib/lxc/{{ item.name | d(item) }}/fstab'
```
Comment (drybjed) [inline ansible/roles/lxc/tasks/main.yml:358]: This change should probably be versioned, LXC 3.x supports the `lxc.mount` variant, and 4.x supports the `lxc.mount.fstab` variant. Also, you might need to update the regexp accordingly.

Comment (stefangweichinger) [reply ansible/roles/lxc/tasks/main.yml:358]: Ack. I will try to come up with some useful suggestion.

**Comment by boucman** [comment]
why do we systematically need cap_sys_admin ?

**Comment by stefangweichinger** [comment]
> why do we systematically need cap_sys_admin ?

I can't tell .. I just found that LXC-4.x seems to need that change (with Debian 11.x "Bullseye").

**Comment by stefangweichinger** [comment]
Did a quick test with 2 target hosts "deb10" and "deb11" running "buster" and "bullseye".
Created 2 lxc-containers "deb10_fstabtest" and "deb11_fstabtest" on them, with host_vars:

```
lxc__containers:

  - name: 'deb10_fstabtest'
    distribution: 'debian'
    release: 'buster'
    ssh: False
    fstab: |
      /srv/shared/lxc-opt opt none bind 0 0
```

and

```
lxc__containers:

  - name: 'deb11_fstabtest'
    distribution: 'debian'
    release: 'bullseye'
    ssh: False
    fstab: |
      /srv/shared/lxc-opt opt none bind 0 0
```

# result

```
deb10 # tail -1 /var/lib/lxc/deb10_fstabtest/config
lxc.mount = /var/lib/lxc/deb10_fstabtest/fstab

deb11 # tail -1 /var/lib/lxc/deb11_fstabtest/config 
lxc.mount.fstab = /var/lib/lxc/deb11_fstabtest/fstab
```

**Comment by stefangweichinger** [comment]
https://discuss.linuxcontainers.org/t/lxc-4-0-6-in-debian-11-x-lxc-cap-drop-sys-admin/11100

**Review by drybjed — APPROVED** [review]
Well, there's no news after two weeks, so I guess we can go with this change for now since it looks like this will be required for Debian Bullseye. I'll backport this PR to the stable releases since there's no change for older LXC versions.

**Comment by stefangweichinger** [comment]
ack, thx

**Pull request merged**